scribejava-apis/src/main/java/com/github/scribejava/apis/KeycloakApi.java

```diff
@@ -0,0 +1,60 @@
+package com.github.scribejava.apis;
+
+import com.github.scribejava.apis.openid.OpenIdJsonTokenExtractor;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.extractors.TokenExtractor;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class KeycloakApi extends DefaultApi20 {
+
+    private final String baseUrlWithRealm;
+
+    private static final ConcurrentMap<String, KeycloakApi> INSTANCES = new ConcurrentHashMap<>();
+
+    protected KeycloakApi(String baseUrlWithRealm) {
+        this.baseUrlWithRealm = baseUrlWithRealm;
+    }
+
+    public static KeycloakApi instance() {
+        return instance("http://localhost:8080/", "master");
+    }
+
+    public static KeycloakApi instance(String baseUrl, String realm) {
+        final String defaultBaseUrlWithRealm = composeBaseUrlWithRealm(baseUrl, realm);
+
+        //java8: switch to ConcurrentMap::computeIfAbsent
+        KeycloakApi api = INSTANCES.get(defaultBaseUrlWithRealm);
+        if (api == null) {
+            api = new KeycloakApi(defaultBaseUrlWithRealm);
+            INSTANCES.putIfAbsent(defaultBaseUrlWithRealm, api);
```

Comment:
The error is here as well ;-)
Method putIfAbsent can ignore second param "api" and doesn't put in the map in case another thread will invoke this method in parallel, and method instance wll return another KeycloakApi instance.
will fix it in this manner:
```
alreadyCreatedApi = INSTANCES.putIfAbsent(defaultBaseUrlWithRealm, api);
    if(alreadyCreatedApi!=null) {
        return alreadyCreatedApi;
    }
```